dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/sam/SamPermissionsImpl.java

```diff
@@ -279,19 +288,37 @@ private void initializePermission(Workflow workflow, User user) {
         String encodedPath = encodedWorkflowResource(workflow, resourcesApi.getApiClient());
         try {
             resourcesApi.createResourceWithDefaults(SamConstants.RESOURCE_TYPE, encodedPath);
-
-            final AccessPolicyMembership writerPolicy = new AccessPolicyMembership();
-            writerPolicy.addRolesItem("writer");
-            resourcesApi.overwritePolicy(SamConstants.RESOURCE_TYPE, encodedPath, SamConstants.WRITE_POLICY, writerPolicy);
-
-            final AccessPolicyMembership readerPolicy = new AccessPolicyMembership();
-            readerPolicy.addRolesItem("reader");
-            resourcesApi.overwritePolicy(SamConstants.RESOURCE_TYPE, encodedPath, SamConstants.READ_POLICY, readerPolicy);
+            addPolicy(resourcesApi, encodedPath, SamConstants.WRITE_POLICY);
+            addPolicy(resourcesApi, encodedPath, SamConstants.READ_POLICY);
         } catch (ApiException e) {
-            throw new CustomWebApplicationException("Error initializing permissions", e.getCode());
+            if (e.getCode() == HttpStatus.SC_CONFLICT) {
+                handleConflict(encodedPath);
```

Comment:
Not entirely clear to me, is a conflict:
1) A user re-requesting initialization of permissions  (i.e. foo@gmail.com has initialized permissions, foo@gmail.com self-destructed, foo@gmail.com was re-created and wants to re-initialize) 
2) A user requesting initialization of permissions but another user has previously initialized it  (foo@gmail.com is trying to initialize but goop@gmail.com already did it) 

Or both? Or perhaps more to the point, will this re-initialize or ignore correctly in 1) and show the message for 2) or both?

Reply:
Case 1 should not get in here; this method is only called if the [resource doesn't exist](https://github.com/ga4gh/dockstore/blob/feature/1713/initpermissions/dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/sam/SamPermissionsImpl.java#L135).

The 409 happens for case 2 only. In that case, admin intervention would be required to resolve this, because another user already owns the resource.

I'll add a comment to make this clearer.